cpp/oneapi/dal/test/engine/dataframe_common.hpp

```diff
@@ -0,0 +1,269 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <any>
+#include <string>
+#include <vector>
+#include <memory>
+#include <optional>
+
+#include "oneapi/dal/array.hpp"
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/test/engine/common.hpp"
+
+namespace oneapi::dal::test::engine {
+
+#define GENERATE_DATAFRAME(...) \
+    GENERATE(as<oneapi::dal::test::engine::dataframe_builder>{}, __VA_ARGS__).build()
+
+enum class table_kind { homogen };
+
+enum class table_float_type { f32, f64 };
+
+class table_id {
+public:
+    template <typename Float>
+    static table_id homogen() {
+        static_assert(dal::detail::is_floating_point<Float>());
+        if constexpr (std::is_same_v<Float, float>) {
+            return table_id{ table_kind::homogen, table_float_type::f32 };
```

Comment:
Why you can't use `homogen_table::kind()` instead of creating your own enum `table_kind`?

https://github.com/rlnx/daal/blob/test-engine/cpp/oneapi/dal/table/homogen.hpp#L57
https://github.com/rlnx/daal/blob/test-engine/cpp/oneapi/dal/table/common.hpp#L144

Reply:
I expect `table::kind()` and `te::table_kind` would diverge. For example, we could create `homogen_row_major` or `homogen_column_major` or custom table type that defined in `test::engine` (e.g. mock).

Reply:
Arguable, but Ok, let's keep this version for now. This is probably the indicator that `table::kind` is useless

Reply:
`table::kind` helps you to determine type of table at runtime, this is the main use case we discussed.